### PR TITLE
fix(ui): Improve detection of specific locale for date formats

### PR DESCRIPTION
**Pull request by dvail**
### Description

Fixes detection of user locales by explicitly checking `navigator.languages` instead of using `undefined` and relying on auto-detection. I have found sources online that state that the resolution of locale when using `undefined` does not always select the first option, leading to unexpected results. Unfortunately I have not found any primary sources in the spec or implementation that confirms this, but I can confirm with the following behavior when setting my browser to use `en-GB` as the default language.

```
navigator.languages
// ['en-GB', 'en']

let fmt = Intl.DateTimeFormat(undefined)
fmt.resolvedOptions().locale
// 'en-US'
```

This behavior results in US formatting still being used for other English locales, such as GB and AU.

In addition, for extra clarity in the case where month/day are _still_ ambiguous, I've switched to display the month short version (3 characters) instead of numeric. This does not take up any additional space, and increases the clarity of the overall date format.

### User-facing documentation

- [ ] [CHANGELOG.md](https://github.com/stackrox/stackrox/blob/master/CHANGELOG.md) is updated **OR** update is not needed
- [ ] [documentation PR](https://spaces.redhat.com/display/StackRox/Submitting+a+User+Documentation+Pull+Request) is created and is linked above **OR** is not needed

### Testing and quality

- [ ] the change is production ready: the change is [GA](https://github.com/stackrox/stackrox/blob/master/PR_GA.md), or otherwise the functionality is gated by a [feature flag](https://github.com/stackrox/stackrox/blob/master/pkg/features/README.md)
- [ ] CI results are [inspected](https://docs.google.com/document/d/1d5ga073jkv4CO1kAJqp8MPGpC6E1bwyrCGZ7S5wKg3w/edit?tab=t.0#heading=h.w4ercgtcg0xp)

#### Automated testing

- [ ] added unit tests
- [ ] added e2e tests
- [ ] added regression tests
- [ ] added compatibility tests
- [ ] modified existing tests

#### How I validated my change

image

### CODE REVIEW

**Comment by dvail** [comment]
This change is part of the following stack:

- #15303 ◀

<sub>Change managed by [git-spice](https://abhinav.github.io/git-spice/).</sub>

File: ui/apps/platform/src/Components/CompoundSearchFilter/components/ConditionDate.tsx
[the remark this answers is not in the document]
Comment (pedrottimark) [reply ui/apps/platform/src/Components/CompoundSearchFilter/components/ConditionDate.tsx:1]: Aha, let’s see if we can follow up early in 4.9 effort to keep up the momentum.

For reference, **EPSS probability** search filter converts between frontend and backend values: #13949

File: ui/apps/platform/src/Components/CompoundSearchFilter/components/ConditionDate.tsx [hunk 1]
```diff
@@ -16,6 +16,10 @@ export type ConditionDateProps = {
     onSearch: (value: ConditionDate) => void;
 };
 
+function dateFormat(date: Date): string {
+    return format(date, 'MM/DD/YYYY');
+}
```
Comment (pedrottimark) [inline ui/apps/platform/src/Components/CompoundSearchFilter/components/ConditionDate.tsx:21]: If I understand correctly from your comment, this is temporary solution, therefore better to encapsulate where it is needed, but otherwise we would import the helper function from dateUtils.ts file?

Comment (dvail) [reply ui/apps/platform/src/Components/CompoundSearchFilter/components/ConditionDate.tsx:21]: Yes, temporary, but I'm inclined to change this to use a general `YYYY-MM-DD` date format for clarity instead, and convert between that and the `MM/DD/YYYY` format the backend expects. 

I am concerned that attempting to detect the format the user is using and do the conversion ourselves will lead to edge cases or difficult to test bugs for different locales that I would rather avoid. Since we are using ISO 8601 elsewhere in the UI, I don't think this is too inconsistent, and the user does still have the option of using the date picker component instead of typing the date in manually.
